osu.Game.Rulesets.Mania/Beatmaps/StageDefinition.cs

```diff
@@ -16,12 +16,40 @@ public struct StageDefinition
         /// </summary>
         public int Columns;
 
+        /// <summary>
+        /// Whether the SpecialStyle flag is set.
+        /// </summary>
+        public bool SpecialStyle;
+
+        /// <summary>
+        /// Whether this is a special style map.
+        /// </summary>
+        /// <remarks>
+        /// Aside from 6K and 8K, special style is almost never used in other
+        /// keymodes, so this function always gives false for non-6/8K modes.
+        /// This also matches stable behavior of only allowing configuration
+        /// of special style in 6/8K.
```

Comment:
"almost never" is not "never" - what does this intend to say? If it *does* get used in other keymodes then it should be supported.

I also don't think this should be limited to 6K and 8K even if stable does ignore the config value for other modes. The less magic conditionals, the better, in my view.